examples/sri-1/components/workgroup/workgroup/workgroup.ts

```diff
@@ -0,0 +1,95 @@
+import { Security } from '../policy/security'
+import { Privacy } from "../policy/privacy"
+
+export class Workgroup implements IWorkgroup {
+    id: string;
+    name: string
+    administrator: BpiSubject[] = []; //[R238] A workgroup MUST have at least one administrator.
+    securityPolicy: Security[] = []; //[R236] There MUST be at least one BPI Subject role that has the authorization to create a workgroup. //[239] A workgroup MUST have at least one security policy. Note that a security policy consists of authentication and authorization rules for the workgroup participants. Note also that one or more workgroup administrators define the workgroup security policy.
+    privacyPolicy: Privacy[] = []; //[R240] A workgroup MUST have at least one security policy. A privacy policy consists of the data visibility rules for each participant.
+    participants: BpiSubject[] = [];
+    worksteps: Workstep[] = [];
+    workflows: Workflow[] = [];
+
+    constructor(id: string, name: string, administrators: BpiSubject[], participants: BpiSubject[], securityPolicy: Security, privacyPolicy: Privacy ) {
+        this.id = id;
+        this.name = name;
+        this.administrator.push(administrators); //[R238] A workgroup MUST have at least one administrator.
+        this.participants.push(participants); //[R237] A workgroup MUST consist of at least one BPI Subject participant.
+        this.securityPolicy.push(securityPolicy); //[239] A workgroup MUST have at least one security policy. Note that a security policy consists of authentication and authorization rules for the workgroup participants. Note also that one or more workgroup administrators define the workgroup security policy.
+        this.privacyPolicy.push(privacyPolicy); //[R240] A workgroup MUST have at least one security policy. A privacy policy consists of the data visibility rules for each participant.
+    }
+
+    //TODO Implement checks for actions bound by security and privacy policies
+    addParticipant(bpiSubject: BpiSubject) { //[R241] A workgroup administrator MUST be able to perform at minimum the following functions
+        return  this.participants.push(bpiSubject);
+    };
+
+    addParticipants(bpiSubjects: BpiSubject[]) { //[R241] A workgroup administrator MUST be able to perform at minimum the following functions
+        bpiSubjects.forEach(subject => {
+            this.participants.push(subject);
+        });
+    }
+
+    getParticipantById(id: string): BpiSubject {
+        return this.participants.find(subject => subject.id == id);
+    };
+
+    getAllParticipants(): BpiSubject[] {
+        return this.participants;
+    };
+
+    updateParticipant(id: string, ...update:any): BpiSubject {}; //TODO write logic for dynamic update #485
+
+    removeParticipant(id: string): BpiSubject { //[R241] A workgroup administrator MUST be able to perform at minimum the following functions:
+        return this.participants = this.participants.filter(subject => subject.id != id);
```

Comment:
I think it would be better that we first check if the provided id exists, instead of reassigning the filtered results each time